src/body/body.rs

```diff
@@ -289,19 +279,16 @@ impl Body {
 
     #[cfg(feature = "http1")]
     pub(super) fn take_full_data(&mut self) -> Option<Bytes> {
-        if let Kind::Once(ref mut chunk) = self.kind {
-            chunk.take()
-        } else {
-            None
-        }
+        None
     }
 }
 
+#[cfg(feature = "ffi")]
 impl Default for Body {
-    /// Returns `Body::empty()`.
+    /// Returns `Body::ffi()`.
     #[inline]
     fn default() -> Body {
-        Body::empty()
+        Body::ffi()
```

Comment:
Does it make sense to use `ffi()` as default?

Reply:
No, I think we'll need to remove the Default impl as well.

Reply:
Ok, if so, we need to address the usage like the following:

```rust
DecodedLength::ZERO => Body::empty(),
```

Reply:
I think it should use this: https://github.com/hyperium/http-body/blob/master/http-body-util/src/empty.rs

Reply:
An inline variant is fine, we can't publicly expose those types since they're not stable.

Reply:
> I think it should use this: https://github.com/hyperium/http-body/blob/master/http-body-util/src/empty.rs

a trait like `Dispatch` requires using the `Body` struct, so I think returning `http_body_util::Empty` here doesn't work:

```rust
pub(crate) trait Dispatch {
    fn recv_msg(&mut self, msg: crate::Result<(Self::RecvItem, Body)>) -> crate::Result<()>;
}
```

> An inline variant is fine, we can't publicly expose those types since they're not stable.

Can you explain a bit? I don't know how to return an empty body here.